multichain-aggregator/multichain-aggregator-logic/src/repository/addresses.rs

```diff
@@ -75,21 +68,112 @@ where
                 .to_owned(),
         )
         .materialized(is_cte_materialized)
-        .table_name(addresses_cte_iden.clone())
+        .table_name(name)
+        .to_owned()
+}
+
+pub async fn uniform_chain_search<C>(
```

Comment:
maybe it's me but it's not obvious: what's the difference between this and `list`?